src/test/ui/parser/recover-for-loop-parens-around-head.stderr

```diff
@@ -0,0 +1,27 @@
+error: expected one of `)`, `,`, or `@`, found `in`
+  --> $DIR/recover-for-loop-parens-around-head.rs:10:16
+   |
+LL |     for ( elem in vec ) {
+   |                ^^ expected one of `)`, `,`, or `@` here
+
+error: unexpected closing `)`
+  --> $DIR/recover-for-loop-parens-around-head.rs:10:23
+   |
+LL |     for ( elem in vec ) {
+   |         --------------^
+   |         |
+   |         opening `(`
+   |         help: remove parenthesis in `for` loop: `elem in vec`
```

Comment:
I'm not entirely happy with the wording:

- The span pointing at the opening brace is slightly redundant.
- The issue is already known to be the superfluous parenthesis, so it thee maim message could be `incorrect parenthesis surrounding for head` or similar (probably something more user friendly).
- The primary span points at the closing parenthesis but it doesn't have a label.
- Suggestions that have very subtle changes sometimes make people not pay attention to the text, in this case people might not realize the lack of parenthesis in the suggested code.
- We have two errors pointing at the problem here (the prior "`in` not expected" error, but it's ok emitting both for now)

IMO something like the following would be better:

```
error: unnecessary parenthesis surroundinig `for` loop head
  --> $DIR/recover-for-loop-parens-around-head.rs:10:23
   |
LL |     for ( elem in vec ) {
   |         ^^^^^^^^^^^^^^^ help: remove parenthesis in `for` loop head
```

Reply:
Filed https://github.com/rust-lang/rust/issues/63113 for following up.